app/main/views/suppliers.py

```diff
@@ -381,7 +372,8 @@ def edit_supplier_declaration_section(supplier_id, framework_slug, section_id):
     '/suppliers/<string:supplier_id>/edit/declarations/<string:framework_slug>/<string:section_id>',
     methods=['POST'])
 def update_supplier_declaration_section(supplier_id, framework_slug, section_id):
-    supplier = data_api_client.get_supplier(supplier_id)['suppliers']
+    # Supplier must exist.
+    data_api_client.get_supplier(supplier_id)
```

Comment:
This feels really weird to me for a couple reasons.

1) It would be nice to catch this exception and 404 a little more explicitly, rather than relying on the API client to terminate the request through an invisibly-handled HTTPError. (Not really anything to do with your flake8 stuff, just a side thought)
2) Reading this line of code is confusing. We get something and don't store it or use it for anything; it would be easy to assume this line of code isn't even needed. It's also less consistent with surrounding code. Personally I'd rather we just keep the unused reference around, but I guess you'll disagree with that. In which case, a much better comment should be fine.